common/flatpak-transaction-log.c

```diff
@@ -0,0 +1,287 @@
+  json_object_set_string_member (object, "origin", origin);
+
+  return base;
+}
+
+/**
+ * flatpak_transaction_log_commit_deploy_install_event:
+ * @log: a #FlatpakTransactionLog
+ * @ref: a decomposable Flatpak ref
+ * @origin: the origin of the flatpak to be installed, whether it is from
+ *          a bundle or from a remote.
+ * @cancellable: a #GCancellable
+ * @error: a #GError
+ *
+ * Write a "deploy-install" event to the transaction log. The "ref" and
+ * "origin" arguments will be included in the event description as properties.
+ *
+ * Returns: %TRUE if writing the event succeeded.
+ */
+gboolean
+flatpak_transaction_log_commit_deploy_install_event (FlatpakTransactionLog *log,
+		                                                 const gchar           *ref,
+		                                                 const gchar           *origin,
+		                                                 GCancellable          *cancellable,
+		                                                 GError                **error)
+{
+  g_autoptr(JsonNode) event = deploy_install_event (ref, origin);
+
+  return append_node_to_log_file (log->path, event, cancellable, error);
+}
+
+static JsonNode *
+uninstall_event (const gchar *ref)
+{
+  JsonNode *base = build_base_event ("uninstall");
+  JsonObject *object = json_node_get_object (base);
+
+  json_object_set_string_member (object, "ref", ref);
+
+  return base;
+}
+
+
+/**
+ * flatpak_transaction_log_commit_uninstall_event:
+ * @log: a #FlatpakTransactionLog
+ * @ref: a decomposable Flatpak ref
+ * @origin: the origin of the flatpak to be installed, whether it is from
+ *          a bundle or from a remote.
+ * @cancellable: a #GCancellable
+ * @error: a #GError
+ *
+ * Write an "install" event to the transaction log. The "ref" arguments will be
+ * included in the event description as properties.
+ *
+ * Returns: %TRUE if writing the event succeeded.
+ */
+gboolean
+flatpak_transaction_log_commit_uninstall_event (FlatpakTransactionLog *log,
+                                                const gchar           *ref,
+                                                GCancellable          *cancellable,
+                                                GError                **error)
+{
+  g_autoptr(JsonNode) event = uninstall_event (ref);
+
+  return append_node_to_log_file (log->path, event, cancellable, error);
+}
+
+static void
+flatpak_transaction_log_class_init (FlatpakTransactionLogClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->finalize = flatpak_transaction_log_finalize;
+  object_class->get_property = flatpak_transaction_log_get_property;
+  object_class->set_property = flatpak_transaction_log_set_property;
+
+  /**
+   * FlatpakTransactionLog:path:
+   *
+   * The absolute path, on disk, where the transaction log will be written to.
+   * If the file indicated by the path, or any of its parent directories do
+   * not exist, they will be created.
```

Comment:
Missing a `Since` line in the documentation comment. (`Since: 0.10.0`)

Same applies to the other new APIs here which should have gtk-doc comments.